app/controllers/common_attribute_states_controller.rb

```diff
@@ -43,25 +40,26 @@ def create
 
   def update_common_attribute_states
     authorize CommonAttributeState
+    @common_attribute_states = []
     common_attribute_state_params.each do |cas_params|
+      params = cas_params.except(:common_attribute_state_id)
       common_attribute_state = CommonAttributeState.find(cas_params[:common_attribute_state_id])
-      # raise ActiveRecord::Rollback unless common_attribute_state.update(cas_params)
-      unless common_attribute_state.update(cas_params.except(:common_attribute_state_id))
-        render :edit, status: :unprocessable_entity
-        return
-      end
+      record_to_update = {:record => common_attribute_state, :params => params}
+      @common_attribute_states.push(record_to_update)
     end
 
-    unless @room.update(last_time_checked: DateTime.now)
-      flash.now['alert'] = "Error updating room record"
-      return
+    ActiveRecord::Base.transaction do
+      @common_attribute_states.each do |cas|
+        raise ActiveRecord::Rollback unless cas[:record].update(cas[:params])
+      end
+      raise ActiveRecord::Rollback unless @room.update(last_time_checked: DateTime.now)
     end
 
-    # if @common_attribute_states.all?(&:persisted?)
-    redirect_to redirect_rover_to_correct_state(room: @room, room_state: @room_state, step: "common_attributes", mode: "edit")
-    # else
-    #   render :edit, status: :unprocessable_entity
-    # end
+    if @room_state.common_attribute_states.all?(&:persisted?)
+      redirect_to redirect_rover_to_correct_state(room: @room, room_state: @room_state, step: "common_attributes", mode: "edit")
+    else
+      render :edit, status: :unprocessable_entity
+    end
   end
```

Comment:
Line 58 here will always resolve to `true`. If we want to check the actual common_attribute_states, we need to use the actual records we updated (stuff in @common_attribute_states). Also, if there are ever any errors, the page will not be able to show them since @common_attribute_states has a different structure.

This solves all the problems (i think?):

```
def update_common_attribute_states
  authorize CommonAttributeState

  @common_attribute_states = common_attribute_state_params.map do |cas_params|
    params = cas_params.except(:common_attribute_state_id)
    common_attribute_state = CommonAttributeState.find(cas_params[:common_attribute_state_id])
    common_attribute_state.assign_attributes(params)
    common_attribute_state
  end

  transaction = ActiveRecord::Base.transaction do
    @common_attribute_states.each do |cas|
      raise ActiveRecord::Rollback unless cas.save
    end
    raise ActiveRecord::Rollback unless @room.update(last_time_checked: DateTime.now)
    true
  end

  if transaction
    redirect_to redirect_rover_to_correct_state(room: @room, room_state: @room_state, step: "common_attributes", mode: "edit")
  else
    render :edit, status: :unprocessable_entity
  end
end
```